vulkan/vulkan.hpp

```diff
@@ -2868,13 +2868,20 @@ namespace VULKAN_HPP_NAMESPACE
         dst[I-1] = src[I-1];
         ConstExpression1DArrayCopy<T, N, I - 1>::copy(dst, src);
       }
+
+      VULKAN_HPP_CONSTEXPR_14 static void copy(T dst[N], const T src[N]) VULKAN_HPP_NOEXCEPT
+      {
+        dst[I - 1] = src[I - 1];
+        ConstExpression1DArrayCopy<T, N, I - 1>::copy(dst, src);
```

Comment:
Can't you use for loop? Don't think reverse copy is the best way to copy on every platform. Also multiple template instantiations slowdown compilation time.

Reply:
Unfortunately, there's no `for constexpr` (yet). So, to have that copy as a constexpr (with C++14), we have to got that route.

Reply:
You don't need `for constexpr`. You just can use `for` in `constexpr` (special)function.

Reply:
If can't, you'll get compile error.

Reply:
This kind of recursive loop unrolling works well with small sized arrays, but copying large buffer with this will probably blow up code size and decrease performance. Should be discussed on different issue or PR I guess.